web-api/src/business/useCaseHelper/trialSessions/trialSessionCalendaring/generateCalendar.ts

```diff
@@ -0,0 +1,338 @@
+import {
+  CalendaringConfig,
+  ProspectiveTrialSession,
+  ScheduledTrialSession,
+} from './createProspectiveTrialSessions';
+import { CaseCountsAndSessionsByCity } from '@web-api/business/useCaseHelper/trialSessions/trialSessionCalendaring/getDataForCalendaring';
+import {
+  Constraint,
+  checkConstraints,
+} from '@web-api/business/useCaseHelper/trialSessions/trialSessionCalendaring/constraints';
+import {
+  FORMATS,
+  createDateAtStartOfWeekEST,
+} from '@shared/business/utilities/DateHandler';
+import { RawTrialSession } from '@shared/business/entities/trialSessions/TrialSession';
+import {
+  SESSION_TYPES,
+  TRIAL_CITY_STRINGS,
+} from '@shared/business/entities/EntityConstants';
+import {
+  WASHINGTON_DC_NORTH_STRING,
+  WASHINGTON_DC_SOUTH_STRING,
+  WASHINGTON_DC_STRING,
+  sortObjectByKey,
+} from '@web-api/business/useCases/trialSessions/generateSuggestedTrialSessionCalendarInteractor';
+
+export type CalendarState = {
+  sessionCountPerWeek: Record<string, number>;
+  sessionScheduledPerCityPerWeek: Record<string, Set<string>>;
+  reservedWeekOfLocationIntersection: Record<string, string[]>;
+  sessionCountPerCity: Record<string, number>;
+};
+
+export const generateCalendar = ({
+  calendaringConfig,
+  caseCountsAndSessionsByCity,
+  constraints,
+  specialSessions,
+  weeksToLoop,
+}: {
+  specialSessions: RawTrialSession[];
+  caseCountsAndSessionsByCity: CaseCountsAndSessionsByCity;
+  weeksToLoop: string[];
+  constraints: Constraint[];
+  calendaringConfig: CalendaringConfig;
+}): {
+  caseCountsAndSessionsByCity: CaseCountsAndSessionsByCity;
+  userMessages: string[];
+} => {
+  const calendarState = setupCalendarState(weeksToLoop);
+
+  sortObjectByKey(caseCountsAndSessionsByCity, (a, b) => {
+    const aNotVisited =
+      caseCountsAndSessionsByCity[a].prospectiveSessions[0]
+        ?.cityWasNotVisitedInLastTwoTerms || false;
+    const bNotVisited =
+      caseCountsAndSessionsByCity[b].prospectiveSessions[0]
+        ?.cityWasNotVisitedInLastTwoTerms || false;
+
+    return aNotVisited === bNotVisited ? 0 : aNotVisited ? -1 : 1;
+  });
+
+  let userMessages: string[] = [];
+  // special sessions handled ahead of all reg, small
+  specialSessions
+    .sort((a, b) => {
+      // eslint-disable-next-line @miovision/disallow-date/no-new-date
+      return +new Date(a.startDate) - +new Date(b.startDate);
```

Comment:
Any reason not to use `createISODateString` to normalize time zones and then compare those lexicographically ordered strings to avoid the new Date? (Or something like that.)